docs/experimental/ticket-auction.md

```diff
@@ -0,0 +1,258 @@
+
+Every NFT created with the wallet also includes some token metadata, following the standard specified for FA2 in [TZIP-12](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-12/tzip-12.md#token-metadata-values) in which token metadata is stored as a Michelson value of type `(map string bytes)`.
+
+### Originations
+
+#### Alice
+
+Alice originates her nft-wallet contract and initializes herself as the admin, the wallet to have no tickets, the ticket_id that will be incremented after every NFT is minted to be 0, and no ticket metadata yet as the wallet does not contain tickets.
+
+```sh
+$ tezos-client originate contract nft-wallet transferring 0 from alice \
+       running "$(pwd)/michelson/nft_wallet.tz" \
+        --init "Pair \"tz1LrQB9HrJcUaD9NKEvV65tnaiU8trPXwmt\" (Pair {} (Pair 0 {}))" \
+        --burn-cap 0.305
+
+        { parameter
+                    (or (or (or (pair %auction
+                                   (contract %destination
+                                      (pair (nat %opening_price)
+                                            (pair (nat %set_reserve_price)
+                                                  (pair (timestamp %set_start_time) (pair (int %set_round_time) (ticket %ticket nat))))))
+                                   (pair (nat %opening_price)
+                                         (pair (nat %reserve_price)
+                                               (pair (timestamp %start_time) (pair (int %round_time) (nat %ticket_id))))))
+                                (nat %burn))
+                            (or (map %mint string bytes) (ticket %receive nat)))
+                        (pair %send (contract %destination (ticket nat)) (nat %ticket_id))) ;
+                  storage
+                    (pair (address %admin)
+                          (pair (big_map %tickets nat (ticket nat))
+                                (pair (nat %current_id) (big_map %token_metadata nat (pair nat (map string bytes)))))) ;
+                  code {...}
+
+New contract KT1EAMUQC1yJ2sRPNPpLHVMGCzroYGe1C1ea originated.
+
+```
+#### Bob
+
+Bob originates his nft wallet and sets himself as the admin
+
+```sh
+$ tezos-client originate contract nft-wallet-bob transferring 0 from bob \
+        running "$(pwd)/michelson/nft_wallet.tz" \
+         --init "Pair \"tz1bwfmSYqrhUTAoybGdhWBBefsbuhNdcC2Y\" (Pair {} (Pair 0 {}))" \
+         --burn-cap 1
+
+New contract KT1QQukCBULzFu6samB5FpbLNBXmNzArSpTs originated.
+
+```
+## NFT Auction Contract
+
+A separate NFT auction contract can be used to auction off ones NFTs to prospective buyers. We use a Dutch, descending price auction, in which the asking price of the NFT is decreased until a buyer is found. The NFT auction contract has an entrypoint to `configure` the auction by customizing
```

Comment:
a link to a more detailed description of Dutch action may be useful here, such as https://en.wikipedia.org/wiki/Dutch_auction